test/Elastic.Apm.EntityFrameworkCore.Tests/EfCoreDiagnosticListenerTests.cs

```diff
@@ -0,0 +1,107 @@
+using System;
+using System.Data.Common;
+using System.Threading.Tasks;
+using Elastic.Apm.Tests.Mocks;
+using FluentAssertions;
+using Microsoft.Data.Sqlite;
+using Microsoft.EntityFrameworkCore;
+using Xunit;
+
+namespace Elastic.Apm.EntityFrameworkCore.Tests
+{
+	public class EfCoreDiagnosticListenerTests : IDisposable
+	{
+		private readonly ApmAgent _apmAgent;
+		private readonly DbConnection _connection;
+		private readonly FakeDbContext _dbContext;
+		private readonly MockPayloadSender _payloadSender;
+
+		public EfCoreDiagnosticListenerTests()
+		{
+			// default EfCore in-memory isn't relational, this is why we're using sqlite in-memory
+			// https://docs.microsoft.com/en-us/ef/core/miscellaneous/testing/in-memory#inmemory-is-not-a-relational-database
+			_connection = new SqliteConnection("DataSource=:memory:");
+			_connection.Open();
+
+			var options = new DbContextOptionsBuilder<FakeDbContext>()
+				.UseSqlite(_connection)
+				.Options;
+
+			_dbContext = new FakeDbContext(options);
+
+			_payloadSender = new MockPayloadSender();
+			_apmAgent = new ApmAgent(new AgentComponents(payloadSender: _payloadSender));
+			_apmAgent.Subscribe(new EfCoreDiagnosticsSubscriber());
+		}
+
+		public void Dispose()
+		{
+			_connection?.Close();
+			_dbContext?.Dispose();
+			_apmAgent?.Dispose();
+		}
+
+		[Fact]
+		public async Task EfCoreDiagnosticListener_ShouldCaptureException_WhenCommandFailed()
+		{
+			// Arrange + Act
+			await _apmAgent.Tracer.CaptureTransaction("transaction", "type", async transaction =>
+			{
+				try
+				{
+					await _dbContext.Database.ExecuteSqlCommandAsync("SELECT * FROM FakeTable");
+				}
+				catch
+				{
+					// ignore
+				}
+			});
+
+			// Assert
+			_payloadSender.Spans.Count.Should().Be(1);
```

Comment:
I think we are totally fine now, maybe long term it'd be also nice to have some tests that assert on the properties of `_payloadSender.FirstSpan.Context.Db`. I think we have some tests doing so in one of the ASP.NET Core tests, but probably this new project would be a good place to have some basic tests on the db fields.